mrtt-ui/src/views/Auth/SignupForm.js

```diff
@@ -0,0 +1,102 @@
+import { Controller, useForm } from 'react-hook-form'
+import { toast } from 'react-toastify'
+import { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+import axios from 'axios'
+
+import { ButtonCancel, ButtonSubmit } from '../../styles/buttons'
+import { ButtonContainer, RowFlexEnd } from '../../styles/containers'
+import { ErrorText } from '../../styles/typography'
+import { Form, MainFormDiv, SectionFormTitle } from '../../styles/forms'
+import { FormLabel, TextField } from '@mui/material'
+import language from '../../language'
+import LoadingIndicator from '../../components/LoadingIndicator'
+
+const validationSchema = yup.object({
+  email: yup.string().required('Email required'),
+  password: yup.string().required('Password required')
+})
+
+const formDefaultValues = { email: '', password: '' }
+
+const SignupForm = () => {
+  const [isLoading] = useState(false)
+  const [isSubmitError, setIsSubmitError] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const navigate = useNavigate()
+
+  const getBaseUrl = (url) => {
+    const pathArray = url.split('/')
+    const protocol = pathArray[0]
+    const host = pathArray[2]
+    return protocol + '//' + host
+  }
+
+  const authUrl = `${getBaseUrl(process.env.REACT_APP_API_URL)}/users`
+
+  const {
+    control: formControl,
+    handleSubmit: validateInputs,
+    formState: { errors }
+  } = useForm({ resolver: yupResolver(validationSchema), defaultValues: formDefaultValues })
+
+  const signUp = (formData) => {
+    axios
+      .post(authUrl, { user: formData })
+      .then(({ data }) => {
+        setIsSubmitting(false)
+        toast.success(data.message)
+        // navigate('/sites')
```

Comment:
please resist submitting commented code. Not a blocker though since we are moving fast.

Reply:
That's why this is still a draft. As commented in the description, we still need to figure out how to navigate in and out of this signup form.